src/lib/use-color-theme.ts

```diff
@@ -0,0 +1,17 @@
+import { useEffect, useState } from 'react';
+import { classes } from '../components/app.st.css'
+
+export function useColorTheme() {
+  const [theme, setTheme] = useState<'dark' | 'light'>('dark')
+  
+  useEffect(() => {
+    window.matchMedia('(prefers-color-scheme: dark)').addEventListener('change', e => {
```

Comment:
always dispose inside hooks. for example if you would use react HMR then all the hooks will be called twice then you will have two listeners